logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/ephemeral/EphemeralMessageDeletionHandler.kt

```diff
@@ -0,0 +1,84 @@
+package com.wire.kalium.logic.feature.message.ephemeral
+
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.functional.onSuccess
+import com.wire.kalium.util.KaliumDispatcher
+import com.wire.kalium.util.KaliumDispatcherImpl
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.datetime.Clock
+import kotlin.coroutines.CoroutineContext
+
+interface EphemeralMessageDeletionHandler {
+
+    fun startSelfDeletion(conversationId: ConversationId, messageId: String)
+    fun enqueuePendingSelfDeletionMessages()
+}
+
+// TODO:Mateusz: implement failure logic (needs to be discussed with a team)
```

Comment:
No need to sign `TODOs` btw. We can always check git blame 😄.

No need to change now that it's done, tho.

Reply:
done